api/serviceregistry/helper.go

```diff
@@ -0,0 +1,3 @@
+package serviceregistry
+
+type validationErrors []error
```

Comment:
This helper file seems to be useless right now. I think adding some validation checks here could help.